apps/server/src/modules/deletion/repo/deletion-request.repo.spec.ts

```diff
@@ -0,0 +1,279 @@
+				const userId = new ObjectId().toHexString();
+
+				const entity: DeletionRequestEntity = deletionRequestEntityFactory.build({ itemId: userId });
+				await em.persistAndFlush(entity);
+
+				const expectedDeletionRequest = {
+					id: entity.id,
+					domain: entity.domain,
+					deleteAfter: entity.deleteAfter,
+					itemId: entity.itemId,
+					status: entity.status,
+					createdAt: entity.createdAt,
+					updatedAt: entity.updatedAt,
+				};
+
+				return {
+					entity,
+					expectedDeletionRequest,
+				};
+			};
+
+			it('should find the deletionRequest', async () => {
+				const { entity, expectedDeletionRequest } = await setup();
+
+				const result: DeletionRequest = await repo.findById(entity.id);
+
+				// Verify explicit fields.
+				expect(result).toEqual(expect.objectContaining(expectedDeletionRequest));
+			});
+		});
+	});
+
+	describe('findAllItemsByDeletionDate', () => {
+		describe('when there is no deletionRequest for execution', () => {
+			it('should return empty array', async () => {
+				const result = await repo.findAllItemsByDeletionDate();
+
+				expect(result).toEqual([]);
+			});
+		});
+
+		describe('when there are deletionRequests for execution', () => {
+			const setup = async () => {
+				const dateInFuture = new Date();
+				dateInFuture.setDate(dateInFuture.getDate() + 30);
+				const deletionRequestEntity1: DeletionRequestEntity = deletionRequestEntityFactory.build({
+					deleteAfter: new Date(2023, 9, 1),
+				});
+				const deletionRequestEntity2: DeletionRequestEntity = deletionRequestEntityFactory.build({
+					deleteAfter: new Date(2023, 9, 1),
+				});
+				const deletionRequestEntity3: DeletionRequestEntity = deletionRequestEntityFactory.build({
+					deleteAfter: dateInFuture,
+				});
+
+				await em.persistAndFlush([deletionRequestEntity1, deletionRequestEntity2, deletionRequestEntity3]);
+				em.clear();
+
+				return { deletionRequestEntity1, deletionRequestEntity2, deletionRequestEntity3 };
+			};
+
+			it('should find deletionRequests with deleteAfter smaller then today', async () => {
+				const { deletionRequestEntity1, deletionRequestEntity2, deletionRequestEntity3 } = await setup();
+
+				const results = await repo.findAllItemsByDeletionDate();
+
+				const expectedArray = [
+					{
+						id: deletionRequestEntity1.id,
+						domain: deletionRequestEntity1.domain,
+						deleteAfter: deletionRequestEntity1.deleteAfter,
+						itemId: deletionRequestEntity1.itemId,
+						status: deletionRequestEntity1.status,
+						createdAt: deletionRequestEntity1.createdAt,
+						updatedAt: deletionRequestEntity1.updatedAt,
+					},
+					{
+						id: deletionRequestEntity2.id,
+						domain: deletionRequestEntity2.domain,
+						deleteAfter: deletionRequestEntity2.deleteAfter,
+						itemId: deletionRequestEntity2.itemId,
+						status: deletionRequestEntity2.status,
+						createdAt: deletionRequestEntity2.createdAt,
+						updatedAt: deletionRequestEntity2.updatedAt,
+					},
+				];
```

Comment:
Please move it to the `setup()` function.

Reply:
done